ir/src/main/scala/org/scalajs/core/ir/Position.scala

```diff
@@ -19,10 +19,13 @@ final case class Position(
 ) {
   def show: String = s"$line:$column"
 
-  def isEmpty: Boolean = {
-    source.getScheme == null && source.getRawAuthority == null &&
-    source.getRawPath == "" && source.getRawQuery == null &&
-    source.getRawFragment == null
+  def isEmpty: Boolean = isEmptyCached
+
+  private lazy val isEmptyCached: Boolean = {
+    val source1 = source
+    source1.getScheme == null && source1.getRawAuthority == null &&
+      source1.getRawPath == "" && source1.getRawQuery == null &&
+      source1.getRawFragment == null
```

Comment:
That helps? Like, seriously?

If it does, maybe instead try moving `source.getRawPath == ""` as a first test, and the rest of the comparisons in a `slowPath()` helper. Most (all) `Position`s should be non-empty because of a non-empty `getRawPath`, so just that test should be enough to answer `false`, like, all the time.